### PR TITLE
Add detection for codeblocks and replace the triggering message with a…

### PR DESCRIPTION
**Pull request by SketchNI**
… response from the bot containing the original message without the codeblock and a link to the gist.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+TWITCH_API_CALLBACK_URL=
+ALLOWED_CALLBACK_URLS=https://p4nth3rb0t-mainframe.herokuapp.com,https://theclaw.dev
+DB_CONN_STRING=
+NODE_ENV=
+DISCORD_TOKEN=
+
+# This requires the "repo" scope.
+GITHUB_ACCESS_TOKEN=
+# probably?
+GITHUB_USERNAME=whitep4nth3r
```
Comment (stephfh) [inline .env.example:16]: I don't think all the user gists should go into Salma's personal GitHub Gists, to not spam her own profile with them.

Which, I guess, means the bot needs its own GitHub organisation now.

File: src/discord.ts [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
+import Discord, { Message, MessageEmbed, MessageReaction, User } from "discord.js";
 import { config } from "./config";
 import {
   fetchVideoByUserId,
   getActiveBroadcasterStreamByBroadcasterId,
 } from "./utils/twitchUtils";
 import type { PartialUser, TextChannel } from "discord.js";
 import type { DiscordReactionRole, StreamInfo } from "./data/types";
+import crypto from "crypto";
+import axios from 'axios';
```
Comment (stephfh) [inline src/discord.ts:12]:
```suggestion
import axios from "axios";
```

File: .env.example (hunk 1 above)
Comment (SketchNI) [reply .env.example:16]: I'm not sure if organizations can have gists but there's nothing stopping `$communityManager` from creating a "theclawcommunity" account.

File: src/discord.ts [hunk 3]
```diff
@@ -87,6 +94,83 @@ discord.on(
   },
 );
 
+discord.on('message', async (message: Message) => {
```
Comment (stephfh) [inline src/discord.ts:97]:
```suggestion
discord.on("message", async (message: Message) => {
```

Generally the formatting seems a bit off here, not blaming you here, but rather that the repo is missing prettier or an `.editorconfig` locally and simply a little diverging IDE settings. :D 

You could do `npm --save-dev prettier eslint-config-prettier eslint-plugin-prettier` for that whole formatting thing too, so it is not an issue in subsequent PRs at all. 

The repo for The Claw team site has a `.prettierrc` you can copy for the settings. -> https://github.com/whitep4nth3r/theclaw.team/blob/main/.prettierrc.json and for it to play nice with ESLint https://github.com/whitep4nth3r/theclaw.team/blob/main/.eslintrc (minus the next.js parts obviously, which are unnecessary in this case).

**Comment by whitep4nth3r** [comment]
This looks great — but we really need to sort a place to put the gists :)

**Comment by SketchNI** [comment]
With your permission, I could create a `TheClawCommunityGists` just for this. Sadly organisations can’t create gists. 

Or we could roll our own solution

**Comment by whitep4nth3r** [comment]
Do whatever you need!

**Comment by SketchNI** [comment]
Created an account using the aformentioned name. I have an access token ready for if/when you merge and push to production :)

**Comment by SketchNI** [comment]
I think we can be merged now :)

**Pull request merged**